hls-plugin-api/src/Ide/PluginUtils.hs

```diff
@@ -236,7 +236,13 @@ usePropertyLsp kn pId p = do
 extractTextInRange :: Range -> T.Text -> T.Text
 extractTextInRange (Range (Position sl sc) (Position el ec)) s = newS
   where
-    focusLines = take (fromIntegral $ el - sl + 1) $ drop (fromIntegral sl) $ T.lines s
+    -- NOTE: Always append an empty line to the end to ensure there are
```

Comment:
This is a little short - it's okay but it would be helpful to say _why_ we don't have enough lines otherwise. You did all that reasoning in the ticket but now we're losing it all!

Reply:
@michaelpj 
Oh yes, I thought the ticket could be enough. Should I simply give a link to that ticket or briefly describe the issue in the NOTE?

Reply:
Well, how would someone find the ticket? The way I think about it is: what would *you* want to read if you were encountering this code for the first time and you had no idea about the history?